Assets/Scripts/Gameplay/SparkleObject.cs

```diff
@@ -1,17 +1,35 @@
+using Puzzles;
 using UnityEngine;
 using UnityEngine.Events;
 using UnityEngine.EventSystems;
+using Attributes;
 
 namespace Gameplay
 {
     public class SparkleObject : InteractableObject
     {
-        [SerializeField] private UnityEvent onClick;
-    
+        public UnityEvent OnSuccess;
+        public UnityEvent OnFailure;
+        [SerializeField] private bool isActive = true;
+        [Restrict(typeof(IPuzzle))] [SerializeField] private GameObject puzzle;
+        private IPuzzle _puzzle;
+
+        private void Awake()
+        {
+            _puzzle = puzzle.GetComponent<IPuzzle>();
+        }
         public override void OnPointerClick(PointerEventData eventData)
         {
-            onClick.Invoke();
-            Debug.Log(gameObject.name); // TODO : Delete after tests
+            if (!isActive) return;
+
+            _puzzle.OnSuccess = OnSuccess;
+            _puzzle.OnFailure = OnFailure;
+            _puzzle.Start();
+        }
+
+        public void SetActive(bool isActive)
```

Comment:
Rename methodargument name, it same as public "isActive", Something like "newIsActive"